mage_ai/data_cleaner/column_type_detector.py

```diff
@@ -54,143 +57,114 @@ def get_mismatched_row_count(series, column_type):
-                zip_code_feature_names.append(col_name)
-            else:
-                integer_feature_names.append(col_name)
+def infer_object_type(series, kwargs):
+    clean_series = series.apply(lambda x: x.strip(' \'\"') if type(x) is str else x)
+    clean_series = clean_whitespace(clean_series)
+    clean_series = clean_series.dropna()
 
-    number_feature_names = float_feature_names + integer_feature_names
-    binary_feature_names += \
-        [col for col in number_feature_names if df[col].nunique(dropna=False) == 2]
-    binary_feature_names += \
-        [col for col in non_number_feature_names if df[col].nunique(dropna=False) == 2]
-    float_feature_names = [col for col in float_feature_names if col not in binary_feature_names]
-    integer_feature_names = \
-        [col for col in integer_feature_names if col not in binary_feature_names]
+    series_unique = series.nunique(dropna=False)
+    clean_series_unique = clean_series.nunique()
 
-    for col_name in subtract(non_number_feature_names, binary_feature_names):
-        df_drop_na = df[col_name].dropna()
-        if df_drop_na.empty:
-            text_feature_names.append(col_name)
+    exact_dtype = type(clean_series.iloc[0])
+    if exact_dtype is list:
+        mdtype = TEXT
+    elif np.issubdtype(exact_dtype, np.bool_):
+        if clean_series_unique <= 2:
+            mdtype = TRUE_OR_FALSE
         else:
-            matches = pd.to_datetime(df_drop_na, infer_datetime_format=True, errors='coerce')
-            if type(df_drop_na.iloc[0]) is list:
-                text_feature_names.append(col_name)
-            elif matches.count() / len(matches) >= DATETIME_MATCHES_THRESHOLD:
-                datetime_feature_names.append(col_name)
-            elif df_drop_na.nunique() / len(df_drop_na) >= 0.8:
-                text_feature_names.append(col_name)
+            mdtype = CATEGORY
+    elif clean_series_unique <= 2:
+        mdtype = TRUE_OR_FALSE
+    else:
+        if all(clean_series.str.match(REGEX_NUMBER)):
+            if clean_series.str.contains(REGEX_FLOAT_NEW_SYM).sum():
+                mdtype = NUMBER_WITH_DECIMALS
             else:
-                word_count, _ = \
-                    df[col_name].dropna().map(lambda x: (len(str(x).split(' ')), str(x))).max()
-                if word_count > MAXIMUM_WORD_LENGTH_FOR_CATEGORY_FEATURES:
-                    text_feature_names.append(col_name)
+                mdtype = NUMBER
+        else:
+            length = len(clean_series)
+            matches = clean_series.str.match(REGEX_DATETIME).sum()
+            if matches/length >= DATETIME_MATCHES_THRESHOLD:
+                mdtype = DATETIME
+            else:
+                correct_emails = clean_series.str.match(REGEX_EMAIL).sum()
+                correct_phone_nums = clean_series.str.match(REGEX_PHONE_NUMBER).sum()
+                correct_zip_codes = clean_series.str.match(REGEX_ZIP_CODE).sum()
+                if correct_emails/length >= 0.90:
```

Comment:
nit: space before and after '/'. same for correct_phone_nums and correct_zip_codes below

Reply:
Fixed